glean/test/tests/CppCatchTest.hs

```diff
@@ -0,0 +1,54 @@
+{-
+  Copyright (c) Meta Platforms, Inc. and affiliates.
+  All rights reserved.
+
+  This source code is licensed under the BSD-style license found in the
+  LICENSE file in the root directory of this source tree.
+-}
+
+{-# LANGUAGE TypeApplications #-}
+module CppCatchTest (main) where
+
+import Control.Exception
+import qualified Data.Text as Text
+import Test.HUnit
+
+import TestRunner
+
+import Glean.Backend as Backend
+import Glean.Init
+import Glean.Query.Thrift as Thrift
+import Glean.Query.Thrift.Internal
+import qualified Glean.Schema.Cxx1.Types as Cxx
+import Glean.Typed hiding (end)
+import Glean.Types
+
+import TestData
+import TestDB
+
+catchTest :: Test
+catchTest = dbTestCase $ \env repo -> do
+
+  -- Pick some fact ids of type Cxx.Name
+  names <- runQuery_ env repo $ allFacts @Cxx.Name
+  let factId x = Text.pack (show (fromFid (idOf (getId x))))
+
+  -- Now query for it but assert the wrong type. Show raise
+  r <- try $ runQuery_ env repo $ angle @Cxx.Type $
+      "$cxx1.Type " <> factId (head names)
+  print (r :: Either SomeException [Cxx.Type])
+    -- should be caught and have a nice trace
```

Comment:
Am I right that this test always passes? Is it possible to assert to trace rather than just printing?

Reply:
Can you add an assertion on "fact has the wrong type" being in the trace